persistence/src/commonMain/kotlin/com/wire/kalium/persistence/backup/DatabaseImporter.kt

```diff
@@ -54,6 +55,29 @@ class DatabaseImporterImpl(private val sqlDriver: SqlDriver) : DatabaseImporter
         restoreTable("MessageRestrictedAssetContent")
     }
 
+    private fun restoreConversations() {
+        // For some strange reason, when restoring the backup db, all last_read_date fields from backup db are hardcoded to 0. Therefore,
+        // before restoring the conversations, we need to set the lastReadTimestamp of the backup conversations to the latest timestamp of
+        // the current conversations so that they don't show up as unread.
+        sqlDriver.execute(
+            """UPDATE $BACKUP_DB_ALIAS.Conversation
+            |SET last_read_date = (
+            |SELECT Conversation.last_read_date
+            |FROM Conversation
+            |WHERE Conversation.qualified_id = $BACKUP_DB_ALIAS.Conversation.qualified_id
+            |AND Conversation.last_read_date > $BACKUP_DB_ALIAS.Conversation.last_read_date
+            |)
+            |WHERE EXISTS (
+            |SELECT 1
+            |FROM Conversation
+            |WHERE Conversation.qualified_id = $BACKUP_DB_ALIAS.Conversation.qualified_id
+            |AND Conversation.last_read_date > $BACKUP_DB_ALIAS.Conversation.last_read_date
+            |);
+            |INSERT OR IGNORE INTO Conversation SELECT * FROM $BACKUP_DB_ALIAS.Conversation;
+            """.trimMargin()
+        )
+    }
```

Comment:
🤔 

Unit test would be great here.

So easy to change a few column names and break this, as unlike the rest of the queries (defined in `.sq` files), this is not verified in compile time.